src/main/java/org/junit/Assert.java

```diff
@@ -812,6 +812,41 @@ public static void assertNotSame(Object unexpected, Object actual) {
         assertNotSame(null, unexpected, actual);
     }
 
+    /**
+     * Asserts that two {@link CharSequence} instances have the same contents.
+     * If they do not, an {@link AssertionError} is thrown.
+     *
+     * @param expected the expected contents
+     * @param actual the object to compare to
+     */
+    public static void assertContentEquals(String expected, CharSequence actual)  {
+        assertContentEquals(null, expected, actual);
+    }
+
+    /**
+     * Asserts that two {@link CharSequence} instances have the same contents.
+     * If they do not, an {@link AssertionError} is thrown with the given message.
+     *
+     * @param message the identifying message for the {@link AssertionError}
+     * @param expected the expected contents
+     * @param actual the object to compare to
+     */
+    public static void assertContentEquals(String message, String expected, CharSequence actual) {
+        if (expected == null && actual == null) {
```

Comment:
I think we should consider writing it like this:

    if (expected == actual) {
      return;
    }
    if (expected == null || actual == null) {
      failNotEquals(message, expected, actual
    }
    ...

That way, we avoid calling `actual.toString()` when `expected` and `actual` point to the same instance.

That being said, since the method is named "assertContentEquals" (and not "assertEquals") perhaps passing in `null` for `expected` should be an error (resulting in a `NullPointerException`). We don't do that anywhere else in this class, but I thought I'd put the idea out there in case someone might want to argue for it.

Reply:
Signature of methods is not correct. We must have opportunity to compare 2 instances of Char Sequence interface.

Reply:
We decided that the expected value should be a string. According to the CharSequence Javadoc "The result of comparing two objects that implement CharSequence is therefore, in general, undefined"

Reply:
You are right. I was confused because the question is open. Please close topic if we don't need compare 2 CharSequence instances. How I can see actual open issues? Please answer me in private message) thanks